### PR TITLE
feat(core)!: replace TextStyle attribute with StartTextStyle/EndTextStyle events

### PR DESCRIPTION
**Pull request by StephanMeijer**
Replace inline text formatting attribute model with SAX-style paired events in `docspec-core`. Breaking change scoped to `docspec-core`; downstream crate migration is a separate follow-up.

## Summary by CodeRabbit

* **New Features**
  * Introduced granular text styling with support for Bold, Italic, Code, Strikethrough, Underline, Subscript, Superscript, and colored Mark styles.

* **Refactor**
  * Restructured the text styling model to better manage nested styles with proper LIFO ordering and auto-closing behavior at block boundaries.

* **Tests**
  * Added comprehensive test coverage for text style handling, stack behavior, and edge cases.

### CODE REVIEW

File: crates/docspec-core/src/stack.rs [hunk 1]
```diff
@@ -135,6 +139,10 @@ impl<S: EventSink> StackTrackingSink<S> {
     /// Handles any intermediate End event (not `EndDocument`): validates the stack,
     /// auto-closes intervening blocks, pops the target frame, and forwards the event.
     fn handle_end_event(&mut self, event: Event) -> Result<()> {
+        while self.style_stack.pop().is_some() {
+            self.sink.handle_event(Event::EndTextStyle)?;
+        }
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec-core/src/stack.rs]: Addressed in 6b5e9f0: handle_end_event now validates the target block before draining text styles, with a regression test confirming mismatched block ends do not emit EndTextStyle or mutate style state.

**Comment by StephanMeijer** [comment]
Going to dump everything into one crate first.
